### ISSUE TITLE
Can I use a password with a stream

### ISSUE THREAD

**Issue by potnoddle** [litedb-org/LiteDB#2192]
I'm using LiteDB on Xamarin and need to preinstall a db. I also wish to add encryption. However, Xamarin on Android won't allow direct access to a asset file, only via a stream. 

Is it possible to combine a stream with a password?

**Comment by JKamsker** [comment]
Yes. The password is passed through `EngineSettings` (or an `AesStream` wrapper), and the stream has to be seekable. An Android asset stream isn't seekable, and passing it directly fails with `NotSupportedException`, so copy it into a `MemoryStream` first:

```csharp
using LiteDB;
using LiteDB.Engine;

var ms = new MemoryStream();
using (Stream asset = Assets.Open("seed.db")) // Xamarin.Android asset
    asset.CopyTo(ms);

using var db = new LiteDatabase(new LiteEngine(new EngineSettings
{
    DataStream = ms,
    Password = "your-password"
}));

// alternative:
// using var db = new LiteDatabase(new AesStream("your-password", ms));
```

With a `MemoryStream`, all changes stay in memory. If the app needs to write to the pre-installed database, copy the asset once to a file in the app's data folder and open it normally with `Filename=...;Password=...`. All three ways were tested on 5.0.21 and on the 6.x prerelease.

Closing this as answered — if it doesn't cover your case, comment here and we'll reopen it.
